disperser/dataapi/queried_operators_handlers.go

```diff
@@ -204,22 +198,95 @@ func (s *server) probeOperatorPorts(ctx context.Context, operatorId string) (*Op
 	return portCheckResponse, nil
 }
 
-// query operator host info endpoint if available
-func getNodeInfo(ctx context.Context, socket string, operatorId string, logger logging.Logger) {
-	conn, err := grpc.Dial(socket, grpc.WithTransportCredentials(insecure.NewCredentials()))
+func (s *server) getOperatorInfo(ctx context.Context, operatorId string) (*core.IndexedOperatorInfo, error) {
+	operatorInfo, err := s.subgraphClient.QueryOperatorInfoByOperatorId(ctx, operatorId)
+	if err != nil {
+		s.logger.Warn("failed to fetch operator info", "operatorId", operatorId, "error", err)
+		return nil, fmt.Errorf("operator info not found for operatorId %s", operatorId)
+	}
+	return operatorInfo, nil
+}
+
+func (s *server) scanOperatorsHostInfo(ctx context.Context, logger logging.Logger) (*SemverReportResponse, error) {
+	registrations, err := s.subgraphClient.QueryOperatorsWithLimit(context.Background(), 10000)
 	if err != nil {
-		logger.Error("Failed to dial grpc operator socket", "operatorId", operatorId, "socket", socket, "error", err)
-		return
+		return nil, fmt.Errorf("failed to fetch indexed registered operator state - %s", err)
 	}
-	defer conn.Close()
-	client := node.NewDispersalClient(conn)
-	reply, err := client.NodeInfo(ctx, &node.NodeInfoRequest{})
+	deregistrations, err := s.subgraphClient.QueryOperatorDeregistrations(context.Background(), 10000)
 	if err != nil {
-		logger.Info("NodeInfo", "operatorId", operatorId, "semver", "unknown")
-		return
+		return nil, fmt.Errorf("failed to fetch indexed deregistered operator state - %s", err)
+	}
+
+	operators := make(map[string]int)
+
+	// Add registrations
+	for _, registration := range registrations {
+		logger.Info("Operator", "operatorId", string(registration.OperatorId), "info", registration)
+		operators[string(registration.OperatorId)]++
+	}
+	// Deduct deregistrations
+	for _, deregistration := range deregistrations {
+		operators[string(deregistration.OperatorId)]--
+	}
+
+	activeOperators := make([]string, 0)
+	for operatorId, count := range operators {
+		if count > 0 {
+			activeOperators = append(activeOperators, operatorId)
+		}
+	}
+	logger.Info("Active operators found", "count", len(activeOperators))
+
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	numWorkers := 40
+	timeout := 2 * time.Second
+	operatorChan := make(chan string, len(activeOperators))
+	semvers := make(map[string]int)
+	worker := func() {
+		for operatorId := range operatorChan {
+			operatorInfo, err := s.getOperatorInfo(ctx, operatorId)
```

Comment:
Sending a RPC call for each operator individually is wasteful. Current live operators and their sate can be retrieved by a single RPC with `GetOperatorState`: https://github.com/Layr-Labs/eigenda/blob/master/core/state.go#L138

Reply:
OperatorState does not provide socket info

Reply:
It's included: https://github.com/Layr-Labs/eigenda/blob/master/core/state.go#L113

Reply:
Refactored to use IndexedChainState. ~8x speedup